### ISSUE TITLE
Fix the backdrop behind modals

### ISSUE THREAD

**Issue by chasenlehara** [CCALI/A2JViewer#183]
Slack thread for context: https://cali.slack.com/archives/C069V9TTN/p1640023807083900

The darker background (backdrop) doesn’t show behind modals: https://github.com/CCALI/a2jviewer/commit/42addeeed9777de3bd222e8e11784734a5853948

That was changed for this issue: https://github.com/CCALI/a2jviewer/issues/43

This should be resolved so that we have the backdrop and it doesn’t conflict with the debug panel.

**Comment by janebitovi** [comment]
The old 3rd party backdrop couldn't be positioned behind the modal because of where our modal is in DOM vs where the library places their backdrop, so I made a new backdrop built into the modal that's positioned correctly.

aria-hidden state of the modal was 'false' even when the modal was hidden, so I fixed that too

the new backdrop behaves the same as the old one, you can click anywhere to close it etc.

![image](https://user-images.githubusercontent.com/92131113/151038371-28c18633-9493-4a98-a2de-02070d48dabb.png)

Additionally, as a smoke test to call attention to the otherwise easy-to-miss a11y problem with the modal's aria-hidden state, the new backdrop CSS is now tied directly to the state:
![image](https://user-images.githubusercontent.com/92131113/151039337-fdf574f2-39c5-4c8c-9fd2-ff01e755b3d6.png)
if aria-hidden breaks again, so will the backdrop
